lib/grape-apiary/sample_generator.rb

```diff
@@ -2,26 +2,17 @@ module GrapeApiary
   class SampleGenerator
     attr_reader :resource, :root
 
-    delegate :unique_params, to: :resource
+    delegate :unique_params, :name, to: :resource
 
     def initialize(resource)
       @resource = resource
       @root     = resource.key.singularize
     end
 
     def sample(id = false)
-      array = resource.unique_params.map do |param|
-        next if param.name == root
-
-        [param.name, param.example]
-      end
-
-      hash = Hash[array.compact]
-
-      hash = hash.reverse_merge(id: Config.generate_id) if id
-      hash = { root => hash } if Config.include_root
-
-      hash
+      generate_jsonapi_entity_sample_hash
+    rescue
```

Comment:
I think you should specify the error to rescue from here.  I like the way you did this tho ! 👌

Reply:
heh it could be any number of errors, which is why the blanket rescue from StandardError and the fall back to the legacy generator. I'll try to narrow them down, there may only be three or four possible errors.